### PR TITLE
Scalable volumetrics [skip ci]

### PR DESCRIPTION
**Pull request by EvgeniiG**
### Purpose of this PR
Added support for arbitrary resolution scaling of Volumetric Lighting based on the Volume Framework. This means you can control the cost and quality of volumetrics by placing volumes in your scene.

---
### Testing status

**Manual Tests**: Yes. Need to test VR (have not tested).
**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/HDRP%252Fscalable-volumetrics/.yamato%252Fupm-ci-abv.yml%2523all_project_ci_trunk

---
### Comments to reviewers
Also see: https://github.com/Unity-Technologies/ScriptableRenderPipeline/pull/6011

### CODE REVIEW

**Review by JMargevics — APPROVED** [review]
http://go/ihhgq Here's my favro card on this PR.
Tested:
- All light types
- Dynamic resolution
- New scalable volume settings
- Local volumes
- Migration

All looks good 🔥
My two minor nitpicks:
image
Tooltip is rather mouthful.  Summoning @JordanL8 to comment :)

Is it correct UX-wise to hide new volume settings behind Advanced Settings? Many people forget to check there and these settings solve the most popular issues that occur with volumetrics: quality & slices protruding through thin meshes.

**Comment by JordanL8** [comment]
Can the examples be moved to the docs and the tooltip be simplified to **Resolution of the volumetric buffer, along the x-axis and y-axis, relative to the resolution of the frame buffer.**?

File: com.unity.render-pipelines.high-definition/Runtime/Sky/SkyManager.cs [hunk 1]
```diff
@@ -913,7 +913,7 @@ public void RenderSky(HDCamera hdCamera, Light sunLight, RTHandle colorBuffer, R
 
         public void RenderOpaqueAtmosphericScattering(CommandBuffer cmd, HDCamera hdCamera,
                                                       RTHandle colorBuffer,
-                                                      RTHandle volumetricLighting,
+                                                      RenderTexture volumetricLighting,
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Sky/SkyManager.cs:916]: Why this change here? Why not a RTHandle  ?

Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Runtime/Sky/SkyManager.cs:916]: is this ok with dynamic resolution ?

**Comment by sebastienlagarde** [comment]
@EvgeniiG I launched a Yamato to know what is the state of the current branch:

https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics/tree/HDRP%252Fscalable-volumetrics/.yamato%252Fupm-ci-abv.yml%2523all_project_ci_2020.1/1920392/job

But it need to adress feedback from lewis + merge HDRP/Staging which conflict with recent change of constant buffer.

Something to check is that volumetric works correctly after a resize of the screen. thanks

File: com.unity.render-pipelines.high-definition/Runtime/Sky/SkyManager.cs (hunk 1 above)
Comment (EvgeniiG) [reply com.unity.render-pipelines.high-definition/Runtime/Sky/SkyManager.cs:916]: Yes, it should work. Forgot to rename.

**Comment by EvgeniiG** [comment]
Waaaait I wasn't done!

**Comment by sebastienlagarde** [comment]
Arf. Do you prefer i revert merge PR or you will do another PR?

**Comment by EvgeniiG** [comment]
I will create another PR.

**Comment by EvgeniiG** [comment]
> Resolution of the volumetric buffer, along the x-axis and y-axis, relative to the resolution of the frame buffer.

Done, see https://github.com/Unity-Technologies/Graphics/pull/188.

**Comment by EvgeniiG** [comment]
> Can the example in the second sentence be moved to the docs and the tooltip be simplified to **Resolution of the volumetric buffer, along the x-axis and y-axis, relative to the resolution of the frame buffer.**?

Done, see https://github.com/Unity-Technologies/Graphics/pull/188.
